### ISSUE TITLE
Design Plugin Dispatcher

### ISSUE THREAD

**Issue by juliababkina** [pivotal/skenario#86]
Now the Skenario environment has a plugin and calls it when we need to communicate with an autoscaler. The idea is to have the Plugin Dispatcher component in the Skenario environment which allows to separate all logic with plugins interactions from main logic. We consider Plugin Dispatcher as black box which is responsible for plugin lifecycle management and communication with it.

Plugin Dispatcher knows: 
Which plugins we need to connect and with which configuration
 Which plugin we need to send what data
Basically, in Skenario we say Plugin Dispatcher scale_horizontally and it knows which plugins have this method. 

![Screenshot 2020-07-14 at 3 29 13 PM](https://user-images.githubusercontent.com/65898660/87425789-eb5b4a80-c5e6-11ea-8eff-f5583ef38778.png)
